cmp/compare.go

```diff
@@ -6,6 +6,9 @@
 //
 // This package is intended to be a more powerful and safer alternative to
 // reflect.DeepEqual for comparing whether two values are semantically equal.
+// It is intended to only be used in tests, as cmp may panic if it cannot
+// compare the values. It's propensity towards panicking means that it is
+// unsuitable for production environments where a spurious panic may be fatal.
```

Comment:
Maybe also mention that performance is not really a goal?

Reply:
Done.